src/utils.c

```diff
@@ -348,7 +348,7 @@ SEXP coerceAs(SEXP x, SEXP as, SEXP copyArg) {
   if (!isNull(getAttrib(x, R_DimSymbol)))
     error(_("'x' must not be matrix or array"));
   if (!isNull(getAttrib(as, R_DimSymbol)))
-    error(_("'as' must not be matrix or array"));
+    error(_("input must not be matrix or array"));
```

Comment:
If users are hit by this, maybe they'd like a more actionable error. If I do want to `shift()` a matrix, how?

maybe suggest wrapping with `data.frame()` or `data.table()` to convert to a column-wise list and `shift()` that?

Reply:
Oh I see.

In that case it is a reason to error for this problem earlier inside shift directly rather than re-use the error message, WDYT?

Reply:
Yes, that actually makes sense, from the point of providing more tailored error message. Other than that we are just duplicating the logic. Then other functions using coerceAs could raise own errors as well. Possibly some of them already doing it.